File: C#/448.find-all-numbers-disappeared-in-an-array.cs
/*
 * @lc app=leetcode id=448 lang=csharp
 *
 * [448] Find All Numbers Disappeared in an Array
 */

// @lc code=start
using System.Collections.Generic;
using System;
public class Solution {
    public IList<int> FindDisappearedNumbers(int[] nums) {
        HashSet<int> s = new HashSet<int>(nums);
        List<int> ans = new List<int>();
        for(int i = 1;i < nums.Length + 1; i ++){
            if(s.Contains(i) == false){
                ans.Add(i);
            }
        }
        return ans;
    }
}
// @lc code=end

File: Python3/448.find-all-numbers-disappeared-in-an-array.py
#
# @lc app=leetcode id=448 lang=python3
#
# [448] Find All Numbers Disappeared in an Array
#

# @lc code=start
class Solution(object):
    def findDisappearedNumbers(self, nums):
    
        for i in range(len(nums)):
            index=abs(nums[i])-1
            nums[index]= - abs(nums[index])
    	return [i+1 for i in range(len(nums)) if nums[i] > 0]
# @lc code=end

